### PR TITLE
[#14] Add external JSON tileset support

### LINKED ISSUE #14: Add support for loading dependent TSX tilesets
# Context

Versions of Tiled > 1.0 allow you to create tilesets where the tileset is actually stored in a TSX file rather than being embedded which was the default prior to 1.0

# Proposal

Dynamically load dependent TSX tilesets (based on what tilesets are *actually* being used) so that they can be parsed and cached.

### PR DESCRIPTION
**Pull request by kamranayub**
Closes #14 

## Changes

- Plugin now loads external tilesets in JSON format
- Add new `externalTilesetPathAccessor` method to override external tileset paths
- Updated to excalibur 0.13.1

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -28,18 +28,19 @@ export default class TiledResource extends Resource<ITiledMap> {
    protected mapFormat: TiledMapFormat;
 
    public imagePathAccessor: (path: string, ts: ITiledTileSet) => string;
+   public externalTilesetPathAccessor: (path: string, ts: ITiledTileSet) => string;
```
Comment (eonarheim) [inline src/index.ts]: Can you add examples of when you would use `imagePathAccessor` and/or `exteranalTilesetPathAccessor`?

**Comment by kamranayub** [comment]
That should be in Readme, I think I added it!

File: README.md [hunk 2]
```diff
@@ -54,11 +54,11 @@ The dist uses a UMD build and will attach itself to the `Extensions.Tiled` globa
...
 The `TiledResource` loadable will load the map file you specify along with any referenced tile set assets (images). 
 
-### Handling Tile Image Paths
+### Handling Tiled Paths
 
-The image paths loaded will be relative to where the exported file was saved.
+The image paths and external tileset paths loaded will be relative to where the exported file was saved.
 
-If you need to override this behavior, you can set `imagePathAccessor` to a custom function that takes two parameters: path and `ITiledTileSet` data.
+If you need to override this behavior, you can set `imagePathAccessor` or `externalTilesetPathAccessor` to a custom function that takes two parameters: path and `ITiledTileSet` data.
```
Comment (eonarheim) [inline README.md]: I'm not sure what these paths should look like

Comment (kamranayub) [reply README.md]: It's right below in the code example? 💯

File: src/index.ts (hunk 1 above)
Comment (kamranayub) [reply src/index.ts]: Better? 👍

**Pull request merged**